runtime/wasmer.go

```diff
@@ -453,3 +474,14 @@ func decodeToInterface(in []byte, t interface{}) (interface{}, error) {
 	output, err := sd.Decode(t)
 	return output, err
 }
+
+// memory allocator
+var memAllocator FreeingBumpHeapAllocator
+
+// init memory allocator if it hasn't been
+func initMemAllocator(mem *wasm.Memory) {
```

Comment:
I'm not sure about this method of having a global allocator, since we might eventually want multiple runtimes. we should probably put it inside the wasm context data using `instance.SetContextData(data)`

Reply:
I agree, I'd like to reference the allocator via context data, however we've discovered an issue with context data that contains pointers, this causes a `runtime error: cgo argument has Go pointer to Go pointer` error.  I can't find a way around this.

Reply:
Fixed by adding a map of RuntimeCtx to wasmer.go, so that pointer references to allocator can be stored there.